src/plugins/orangehrmRecruitmentPlugin/Controller/ApplicantController.php

```diff
@@ -0,0 +1,317 @@
+        if (!$csrfTokenManager->isValid('recruitment-applicant', $token)) {
+            throw AuthenticationException::invalidCsrfToken();
+        }
+
+        $variables = $request->request->all();
+
+        /** @var UploadedFile $file */
+        $file = $request->files->get(self::PARAMETER_RESUME);
+        $attachment = Base64Attachment::createFromUploadedFile($file);
+        $variables[self::PARAMETER_RESUME] = [
+            'name' => $attachment->getFilename(),
+            'type' => $attachment->getFileType(),
+            'base64' => $attachment->getBase64Content(),
+            'size' => $attachment->getSize(),
+        ];
+        $paramRules = $this->getParamRuleCollection();
+        $paramRules->addExcludedParamKey('_token');
+
+        try {
+            $this->validate($variables, $paramRules);
+        } catch (InvalidParamException $e) {
+            $this->getLogger()->error($e->getMessage());
+            $this->getLogger()->error($e->getTraceAsString());
+        }
+
+        $this->beginTransaction();
+        try {
+            $applicant = new Candidate();
+            $this->setApplicant($applicant, $request);
+            $applicant = $this->getCandidateService()->getCandidateDao()->saveCandidate($applicant);
+            $lastInsertApplicantId = $applicant->getId();
+
+            $applicantHistory = new CandidateHistory();
+            $this->setCommonApplicantHistoryAttributes(
+                $applicantHistory,
+                $lastInsertApplicantId,
+                CandidateService::RECRUITMENT_CANDIDATE_ACTION_ADD
+            );
+            $this->getCandidateService()->getCandidateDao()->saveCandidateHistory($applicantHistory);
+
+            $vacancyId = $variables[self::PARAMETER_VACANCY_ID];
+            if (!is_null($vacancyId)) {
+                $applicantVacancy = new CandidateVacancy();
+                $this->setApplicantVacancy(
+                    $applicantVacancy,
+                    $lastInsertApplicantId,
+                    CandidateService::STATUS_MAP[WorkflowStateMachine::RECRUITMENT_APPLICATION_ACTION_ATTACH_VACANCY],
+                    $vacancyId
+                );
+                $this->getCandidateService()->getCandidateDao()->saveCandidateVacancy($applicantVacancy);
+
+                $applicantHistory = new CandidateHistory();
+                $this->setCommonApplicantHistoryAttributes(
+                    $applicantHistory,
+                    $lastInsertApplicantId,
+                    CandidateService::RECRUITMENT_CANDIDATE_ACTION_ADD
+                );
+                $applicantHistory->getDecorator()->setVacancyById($vacancyId);
+                $this->getCandidateService()->getCandidateDao()->saveCandidateHistory($applicantHistory);
+            }
+
+            $applicantAttachment = new CandidateAttachment();
+            $this->setCandidateAttachment($applicantAttachment, $lastInsertApplicantId, $variables['resume']);
```

Comment:
```suggestion
            $this->setCandidateAttachment($applicantAttachment, $lastInsertApplicantId, $attachment);
```
can pass `Base64Attachment`

Reply:
Done